### DISCUSSION TITLE
Error while loading camera list: 401 (after a successful authentication)

### DISCUSSION THREAD

**Discussion by daynaskully** [danielfernau/unifi-protect-video-downloader#155] [General]
I'm using the legacy (--not-unifi-os) setting on a CKG2+ (v1.1.11) & Protect (v1.13.7)

> Getting camera list
>Successfully authenticated as user ubnt
>Error while loading camera list: 401
>0 files downloaded (0.0 b), 0 files skipped, 0 files failed, 0 files total`

**Comment by danielfernau** [comment]
Hey there!

Do you have a specific reason as to why you're still on this quite old version of CK firmware and Protect app?

401 is an authentication error, which, however, confuses me a bit since it seems to be able to authenticate just fine...? 
The error you're seeing basically means that the tool was able to perform the authentication handshake but immediately afterwards fails to load the camera list due to not being authenticated.

Using the legacy option is correct in this case. You might want to use a hostname ("unifi") instead of the IP address (or vice versa) and double-check that the account you're using for authentication has the correct permissions set in the CloudKey's web interface.

Hope some of this helps resolving your issue; feel free to reach out if you need additional assistance!

**Reply by daynaskully** [reply]
Mainly due to legacy hardware/setup that will break with updates.

However, ip or hostname makes no difference. And the user is the default and only user with all permissions. 

I can use curl to get a proper authorization response/header from my local cli (that's as far as i've gotten in trying to manually call the commands); I'm also using docker to run this -- could this somehow be a date/time related issue with the image?

Any help would be appreciated.

**Reply by danielfernau** [reply]
Ah, I see. Makes sense. Also, sorry for the delayed reply.

Okay, thanks for checking. Does the tool give you any kind of error message in the console?
What exactly do you mean by 'date/time related issue'? Docker should work fine and is the easiest way to run this tool.
Let's see if we can figure out what's going wrong...

**Reply by daynaskully** [reply]
I thought the date/time might be a token expiry issue or something along those lines (I've had issues in the past running docker containers that used some other time/timezone that would cause cookies to expire due to the dates being so far off -- just a thought)

But generally speaking I can set the bearer auth token manually in the reconstructed calls I make, and get lists of cams/details and anything else without issue.

**Reply by danielfernau** [reply]
After digging a bit through the old code, I might have found something... If you look at 

https://github.com/danielfernau/unifi-protect-video-downloader/blob/2165204442d30676f3f48ad445c0211c6e5c00c1/main.py#L136 

you can see that I've used an Authorization header with the request; whereas in the current code at 

https://github.com/danielfernau/unifi-protect-video-downloader/blob/de8ba275973750824799b41975f2bf7aab98dda0/protect_archiver/downloader/get_camera_list.py#L18 

I'm using a TOKEN cookie. I might have to add an additional if-statement there which uses an Authorization header for legacy Protect systems and the TOKEN cookie method for UniFi OS consoles...

**Reply by danielfernau** [reply]
Fixed in https://github.com/danielfernau/unifi-protect-video-downloader/commit/3cd5a074594c0081afd8a86782e3bf170ebbe00a
Have a look if it solves your issue: `docker pull unifitoolbox/protect-archiver:edge` 😃

**Reply by daynaskully** [reply]
Yep that has worked! Thanks for this. 👍

**Reply by danielfernau** [reply]
You're welcome. And thanks for the feedback!
